src/Amazon.AspNetCore.Identity.Cognito/CognitoUserClaimsPrincipalFactory.cs

```diff
@@ -59,5 +61,44 @@ public async Task<ClaimsPrincipal> CreateAsync(TUser user)
             var claimsIdentity = new ClaimsIdentity(claims, IdentityConstants.ApplicationScheme);
             return new ClaimsPrincipal(claimsIdentity);
         }
+
+        /// <summary>
+        /// Internal method to map System.Security.Claims.ClaimTypes to Cognito Standard Attributes
+        /// </summary>
+        /// <param name="claims"></param>
+        private void MapClaimTypesToCognito(List<Claim> claims)
+        {
+            var emailClaim = claims.FirstOrDefault(claim => claim.Type == CognitoAttributesConstants.Email);
```

Comment:
can we add each of these 3 line null/add checks in a method that take a claim and a constant, and return a Claim/null?

Also, we can just add the the list blindly and remove all nulls at the end.

Reply:
Moved the mapping to a Dictionary<string, string>.

Extracted the mapping logic to a small function.